src/lxc/confile.c

```diff
@@ -1758,22 +1758,35 @@ static int set_config_mount_auto(const char *key, const char *value,
 	    { "proc",                    LXC_AUTO_PROC_MASK,   LXC_AUTO_PROC_MIXED                               },
 	    { "proc:mixed",              LXC_AUTO_PROC_MASK,   LXC_AUTO_PROC_MIXED                               },
 	    { "proc:rw",                 LXC_AUTO_PROC_MASK,   LXC_AUTO_PROC_RW                                  },
+
 	    { "sys",                     LXC_AUTO_SYS_MASK,    LXC_AUTO_SYS_MIXED                                },
 	    { "sys:ro",                  LXC_AUTO_SYS_MASK,    LXC_AUTO_SYS_RO                                   },
 	    { "sys:mixed",               LXC_AUTO_SYS_MASK,    LXC_AUTO_SYS_MIXED                                },
 	    { "sys:rw",                  LXC_AUTO_SYS_MASK,    LXC_AUTO_SYS_RW                                   },
+
 	    { "cgroup",                  LXC_AUTO_CGROUP_MASK, LXC_AUTO_CGROUP_NOSPEC                            },
 	    { "cgroup:mixed",            LXC_AUTO_CGROUP_MASK, LXC_AUTO_CGROUP_MIXED                             },
 	    { "cgroup:ro",               LXC_AUTO_CGROUP_MASK, LXC_AUTO_CGROUP_RO                                },
 	    { "cgroup:rw",               LXC_AUTO_CGROUP_MASK, LXC_AUTO_CGROUP_RW                                },
+
 	    { "cgroup:force",            LXC_AUTO_CGROUP_MASK, LXC_AUTO_CGROUP_NOSPEC|LXC_AUTO_CGROUP_FORCE      },
```

Comment:
Hm, is that really necessary. Seems to me that we should only support `cgroup:<mount-type>:force`.

Reply:
I'm not nacking it I'm really just asking. :)

Reply:
@brauner 
https://github.com/lxc/lxc/blob/master/src/lxc/cgroups/cgfsng.c#L2070 why we need judge has_cgns？ if a container has no sys_admin, it can't mount by itself, so I think lxc should mount cgroup for container if it has no sys_admin. WDYT?

What is it the purpose of judge has_cgns，Is it to make /sys/fs/cgroup has the dir of  lxc/containerid ?

On the other hand, cg_mount_in_cgroup_namespace make container has no lxc/containerid in /sys/fs/cgroup,but do_secondstage_mounts_if_needed has lxc/containerid in /sys/fs/cgroup, Is it a issue ?

I add all cgroup options with force, and also cgroup-direct to make both cg_mount_in_cgroup_namespace and do_secondstage_mounts_if_needed has no lxc/containerid in container's /sys/fs/cgroup.